lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp

```diff
@@ -383,6 +383,68 @@ broadenSingleElementStores(StoreInst *storeInst,
   return killInstruction(storeInst, nextII, pass);
 }
 
+//===----------------------------------------------------------------------===//
+//                            Simple ARC Peepholes
+//===----------------------------------------------------------------------===//
+
+static SILBasicBlock::iterator
+eliminateSimpleCopies(CopyValueInst *cvi, CanonicalizeInstruction &pass) {
+  auto next = std::next(cvi->getIterator());
+
+  // Eliminate copies that only have destroy_value uses.
+  SmallVector<DestroyValueInst *, 8> destroys;
+  for (auto *use : getNonDebugUses(cvi)) {
+    if (auto *dvi = dyn_cast<DestroyValueInst>(use->getUser())) {
+      destroys.push_back(dvi);
+      continue;
+    }
+    return next;
+  }
+
+  while (!destroys.empty()) {
+    next = killInstruction(destroys.pop_back_val(), next, pass);
+  }
+
+  auto op = cvi->getOperand();
+  next = killInstAndIncidentalUses(cvi, next, pass);
+  pass.notifyHasNewUsers(op);
```

Comment:
Is it correct to refer to an operand after deleting the instruction?
And, why would we call `notifyNewUsers` here anyway? `killInstruction` already notified, and there's no new users

Reply:
Fixed.